.github/workflows/lint.yml

```diff
@@ -0,0 +1,23 @@
+name: Tests
+
+on:
+  pull_request:
+    paths:
+      - src/**
```

Comment:
Nit: let's also add `.clang-format` itself here, if we end up changing the clang-format definition and CI doesn't check if there's lint for that commit it could easily lead to confusion.